main/mbcs.c

```diff
@@ -0,0 +1,100 @@
+/*
+*   $Id$
+*
+*   Copyright (c) 2015, vim-jp
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License.
+*
+*   This module contains functions for checking multibyte character set.
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#define __USE_GNU
+#include "general.h"  /* must always come first */
+
+#ifdef HAVE_ICONV
+
+#include <stdio.h>
+#include <string.h>
+#include <iconv.h>
+#include <errno.h>
+#include "options.h"
+#include "mbcs.h"
+#include "routines.h"
+
+static iconv_t iconv_fd = (iconv_t) -1;
+
+extern boolean openConverter (char* inputEncoding, char* outputEncoding)
+{
+	if (!inputEncoding || !outputEncoding)
+		return FALSE;
+	iconv_fd = iconv_open(outputEncoding, inputEncoding);
+	if (iconv_fd == (iconv_t) -1)
+	{
+		error (FATAL,
+					"failed openning encoding from '%s' to '%s'", inputEncoding, outputEncoding);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+extern boolean isConverting ()
+{
+	return iconv_fd != (iconv_t) -1;
+}
+
+extern boolean convertString (vString *const string)
+{
+	size_t utf8_len = 0, mbcs_len, utf8_size;
+	char *utf8, *mbcs, *utf8ptr;
+	if (iconv_fd == (iconv_t) -1)
+		return FALSE;
+	mbcs_len = vStringLength (string);
+	utf8_len = mbcs_len * 4;
```

Comment:
This is probably still OK as I don't know an encoding (actually) using more than 4 bytes per code point, but this might warrant a comment